librad/src/git/storage.rs

```diff
@@ -456,12 +452,7 @@ impl<S: Clone> Storage<S> {
 }
 
 impl Storage<NoSigner> {
-    /// Open the `Storage` found at the given [`Path`]'s `git_dir.
-    ///
-    /// The `Storage` must have been initialised with [`Storage::init`] prior to
-    /// calling this method.
-    pub fn open(paths: &Paths) -> Result<Self, Error> {
-        let backend = git2::Repository::open_bare(paths.git_dir())?;
+    pub fn open(backend: git2::Repository) -> Result<Self, Error> {
```

Comment:
One worry I'm having going through this is that the `Repository` is assumed to be opened bare right? I wonder if we should have a newtype around `git2::Repository` and a method `from_paths` which is the smart constructor for constructing that newtype.